scripts/fast-reboot

```diff
@@ -212,8 +209,60 @@ function teardown_control_plane_assistant()
     fi
 }
 
+function setup_reboot_variables()
+{
+    # Kernel and initrd image
+    NEXT_SONIC_IMAGE=$(sonic_installer list | grep "Next: " | cut -d ' ' -f 2)
+    IMAGE_PATH="/host/image-${NEXT_SONIC_IMAGE#SONiC-OS-}"
+    if grep -q aboot_platform= /host/machine.conf; then
+        KERNEL_IMAGE="$(ls $IMAGE_PATH/boot/vmlinuz-*)"
+        BOOT_OPTIONS="$(cat "$IMAGE_PATH/kernel-cmdline" | tr '\n' ' ') SONIC_BOOT_TYPE=${BOOT_TYPE_ARG}"
+    elif grep -q onie_platform= /host/machine.conf; then
+        KERNEL_OPTIONS=$(cat /host/grub/grub.cfg | sed "/$NEXT_SONIC_IMAGE'/,/}/"'!'"g" | grep linux)
+        KERNEL_IMAGE="/host$(echo $KERNEL_OPTIONS | cut -d ' ' -f 2)"
+        BOOT_OPTIONS="$(echo $KERNEL_OPTIONS | sed -e 's/\s*linux\s*/BOOT_IMAGE=/') SONIC_BOOT_TYPE=${BOOT_TYPE_ARG}"
+    else
+        error "Unknown bootloader. ${REBOOT_TYPE} is not supported."
+        exit "${EXIT_NOT_SUPPORTED}"
+    fi
+    INITRD=$(echo $KERNEL_IMAGE | sed 's/vmlinuz/initrd.img/g')
+}
+
+function reboot_pre_check()
+{
+    # Make sure that the file system is normal: read-write able
+    filename="/host/test-`date +%Y%m%d-%H%M%S`"
+    if [[ ! -f ${filename} ]]; then
+        touch ${filename}
+    fi
+    rm ${filename}
```

Comment:
The script as option -e. so it would error out and take the clean up path. But I am unable to test this code path for the reason I stated in the description.

Reply:
OK. I figured out a way to test it. I changed the script to test /host/test/ and mount /host/test as read-only fs:

mount -t tmpfs -r -o size=100M tmpfs /host/test/

host> warm-reboot -v
touch: cannot touch '/host/test/test-20190412-223147': Read-only file system
Fri Apr 12 22:31:47 UTC 2019 warm-reboot failure (1) cleanup ...
Fri Apr 12 22:31:48 UTC 2019 Cancel warm-reboot: code (1)